bin/arm-template-merge

```diff
@@ -126,7 +128,14 @@ async function mergeTemplateWith(target, inFile) {
 }
 
 async function loadTemplate(inFile) {
-  let json = await readFile(inFile, 'utf8');
+  let jsonBuf = await readFile(inFile);
+
+  // Strip the UTF-8 BOM, if present
+  if (jsonBuf.slice(0, 3).equals(utf8BOM)) {
+    jsonBuf = jsonBuf.slice(3);
+  }
+
+  let json = jsonBuf.toString('utf8');
```

Comment:
Should be const as don't change in the code.

Reply:
@poveden https://eslint.org/docs/rules/prefer-const as an error?

Reply:
I think of `const` as it's used in other languages, that is, something that will have its value set before the code even starts running (i.e. compile-time, although that's not the case in JavaScript).

For me, `const` conveys a clear meaning, which is something that always will have the same value, and not something that may have a different value each time I enter some block of code, even if it won't change in a given iteration.

Reply:
This is an usual confusion between inmutable and constant when coming from other languages.
Const in javascript:
- Improves readability, you know that a const cannot be reasigned.
- Improves performance through optimization of Turbofan for Node 8, liftoff for Node 10.

You have a good article about the topic that const is not inmutability here: https://mathiasbynens.be/notes/es6-const

Also, is one of the standards in common sets of eslint rules, like airbnb. 

So I suggest to use it.

Reply:
Oh, it's clear to me that `const` doesn't mean "immutability". Other languages "fake" it by not letting assign objects to a const, but only primitives (or things that can be resolved at compile-time and are immutable, like a string).

Maybe it's just that I'm getting older but, precisely because of that association above, I see `const` as that. And frankly, I'm not the "because AirBnb does it, we should, too" kind of guy. It's, to me, one of those "holy wars" (like spaces vs. tabs on HTML, or the legendary `vi` vs. `emacs`).

The "improves performance" part is really news to me (and surprises me equally), so I promise I will take a deep look at it and reconsider my "crusade-like" stance.